2019Labs/CustomSecurityContent/documentation/lab3_oval.adoc

```diff
@@ -0,0 +1,487 @@
+</def-group>
+----
+
+This way, there won't be copy-paste errors any more!
+
+Let's examine the test scenarios - for example the `wrong_value.fail.sh`.
+Open a new terminal window, and change to the test definitions directory: `cd tests/data/group_system/group_accounts/group_accounts-session/rule_accounts_tmout`
+Then, let's open the `wrong_value.fail.sh` file.
+
+// gedit screenshot
+
+As you can see, the test sets the `TMOUT` value to 1234.
+This is considered wrong, which is correct - as the timeout should be 600, 1234 is longer, therefore less secure.
+
+What about the `correct.pass.sh`?
+Let's check that out and open it in an editor.
+As we can see, this one sets the `TMOUT` value to 600, which is the value defined by the profile.
+
+Let's add another check for a correct value - let's check for timeout of 100.
+In case of a timeout, 100 seconds is more secure than 600 seconds.
+Therefore, the scenario is represents a *supercompliant* case, i.e. when the setting is stricter, than what is demanded, while being in the area of allowed values.
+So let's copy that one, and make a new test scenario out of it.
+Run this command in the terminal in the `rule_accounts_tmout` directory: `cp correct_value.pass.sh supercompliant.pass.sh`
+
+Let's run the tests!
+
+....
+$ ./build_product fedora
+...
+...
+$ sudo tests/test_suite.py rule --container ssg_test_suite --datastream build/ssg-fedora-ds.xml accounts_tmout                                                            
+Setting console output to log level INFO
+INFO - The base image option has been specified, choosing Podman-based test environment.
+INFO - Logging into /home/lab-user/labs/lab3_oval/content/logs/...
+INFO - xccdf_org.ssgproject.content_rule_accounts_tmout
+INFO - Script correct_value.pass.sh using profile xccdf_org.ssgproject.content_profile_ospp OK
+INFO - Script comment.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp OK
+ERROR - Script supercompliant.pass.sh using profile xccdf_org.ssgproject.content_profile_ospp found issue:                                                                                                        
+ERROR - Rule evaluation resulted in fail, instead of expected pass during initial stage
+ERROR - The initial scan failed for rule 'xccdf_org.ssgproject.content_rule_accounts_tmout'.
+INFO - Script line_not_there.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp OK
+INFO - Script wrong_value.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp OK
+....
+
+The message tells us that the `supercompliant.pass.sh` scenario has failed, which was not expected.
+
+
+== OVAL development
+
+
+=== Correct handling of supercompliance
+
+The test cases are accurate, it is the OVAL that needs to be changed.
+Therefore, let's modify it!
```

Comment:
Let's modify the OVAL definition to allow also the values smaller than the troshold.